src/globus_sdk/services/timer/data.py

```diff
@@ -0,0 +1,133 @@
+import datetime
+from typing import Any, Dict, Optional, Union
+
+from globus_sdk.config import get_service_url
+from globus_sdk.services.transfer import TransferData
+from globus_sdk.utils import PayloadWrapper, slash_join
+
+
+class TimerJob(PayloadWrapper):
+    r"""
+    Class for specifying parameters used to create a job in the Timer service. Used as
+    the ``data`` argument in
+    :meth:`create_job <globus_sdk.TimerClient.create_job>`.
+
+    Timer operates through the `Globus Automate API
+    <https://docs.globus.org/globus-automation-services/>`_. Crucially, the
+    ``callback_url`` parameter should always be the URL used to run an action provider.
+    Currently the only supported action provider for this is for Transfer. Thus, users
+    should generally only use the :meth:`~from_transfer_data` method here. Any other
+    usage is meant for internal purposes; proceed with caution!
```

Comment:
Maybe format with
```
.. warning:
    some text
```

so that this is very attention-get-y. Otherwise, users are probably going to try `__init__` and get confused